temporal-testing-junit5/src/main/java/io/temporal/testing/TestWorkflowExtension.java

```diff
@@ -134,6 +141,10 @@ public boolean supportsParameter(
       return false;
     }
 
+    if (includesDynamicWorkflow) {
+      return true;
```

Comment:
Can we add a short javadoc here explaining the logic behind this branch?

Reply:
Same here - rewritten the code to check that only proper workflow interfaces are injectable.